controller/src/main/java/org/dswarm/controller/resources/ExtendedMediumBasicDMPResource.java

```diff
@@ -0,0 +1,193 @@
+
+		switch (format) {
+			case MEDIUM:
+				return getMediumObject(uuid);
+			default:
+				return super.getObject(uuid, format);
+		}
+	}
+
+	/**
+	 * This endpoint returns a list of all objects of the type of the POJO class as JSON representation.
+	 * The format of the objects might either be a full or an abbreviated short variant.
+	 *
+	 * @param format an enum that specifies which format to use
+	 * @return a list of all objects of the type of the POJO class as JSON representation
+	 * @throws org.dswarm.controller.DMPControllerException
+	 */
+	@Override
+	public Response getObjects(final POJOFormat format) throws DMPControllerException {
+
+		switch (format) {
+			case MEDIUM:
+				return getMediumObjects();
+			default:
+				return super.getObjects(format);
+		}
+	}
+
+	private Response getMediumObject(final String uuid) throws DMPControllerException {
+
+		LOG.debug("try to get {} with uuid '{}'", pojoClassName, uuid);
+
+		final POJOCLASSPERSISTENCESERVICE persistenceService = persistenceServiceProvider.get();
+		final POJOCLASS object = persistenceService.getObject(uuid);
+
+		if (object == null) {
+
+			LOG.debug("couldn't find {} '{}'", pojoClassName, uuid);
+
+			return Response.status(Response.Status.NOT_FOUND).build();
+		}
+		LOG.debug("got {} with uuid '{}'", pojoClassName, uuid);
+		if (LOG.isTraceEnabled()) {
+			LOG.trace(" = '{}'", ToStringBuilder.reflectionToString(object));
+		}
+
+		final String objectJSON = serializeMediumObject(object);
+
+		LOG.debug("return {} with uuid '{}'", pojoClassName, uuid);
+		if (LOG.isTraceEnabled()) {
+			LOG.trace(" = '{}'", objectJSON);
+		}
+
+		return buildResponse(objectJSON);
+	}
+
+	private Response getMediumObjects() throws DMPControllerException {
+
+		LOG.debug("try to get all {}s", pojoClassName);
+
+		final POJOCLASSPERSISTENCESERVICE persistenceService = persistenceServiceProvider.get();
+
+		final List<MEDIUMCLASS> objects = persistenceService.getMediumObjects();
+
+		if (objects == null) {
+
+			LOG.debug("couldn't find {}s", pojoClassName);
+			return Response.status(Response.Status.NOT_FOUND).build();
+		}
+
+		if (objects.isEmpty()) {
+
+			LOG.debug("there are no {}s", pojoClassName);
+			return Response.status(Response.Status.NOT_FOUND).build();
+		}
+
+		LOG.debug("got all {}s ", pojoClassName);
+		if (LOG.isTraceEnabled()) {
+			LOG.trace(" = '{}'", ToStringBuilder.reflectionToString(objects));
+		}
+
+		final List<MEDIUMCLASS> dtos = objects.stream()
+				.map(dto -> dto.withHref(createObjectURI(dto.uuid)))
+				.collect(Collectors.toList());
+
+		final String objectsJSON = serializeObject(dtos);
+
+		LOG.debug("return all {}s ", pojoClassName);
+		if (LOG.isTraceEnabled()) {
+			LOG.trace("'{}'", objectsJSON);
+		}
+
+		return buildResponse(objectsJSON);
+	}
+
+	private String serializeMediumObject(final POJOCLASS pojo) throws DMPControllerException {
+
+		final MEDIUMCLASS dto = mediumVersionOf(pojo);
+
+		try {
```

Comment:
Ok. I thought, there were some differences in the error-case log message, but I can use the existing method.